File: common/src/main/java/com/ssblur/scriptor/word/descriptor/focus/inventory/CasterIgnoreTargetedSlotDescriptor.java
package com.ssblur.scriptor.word.descriptor.focus.inventory;

import com.ssblur.scriptor.api.word.Descriptor;
import com.ssblur.scriptor.helpers.targetable.InventoryTargetable;
import com.ssblur.scriptor.helpers.targetable.Targetable;
import com.ssblur.scriptor.word.descriptor.focus.FocusDescriptor;

public class CasterIgnoreTargetedSlotDescriptor extends Descriptor implements FocusDescriptor {
  @Override
  public Cost cost() {
    return new Cost(0, COSTTYPE.ADDITIVE);
  }

  @Override
  public Targetable modifyFocus(Targetable targetable) {
    if(targetable instanceof InventoryTargetable inventoryTargetable)
      inventoryTargetable.setTargetedSlot(-1);
    return targetable;
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/word/descriptor/focus/inventory/CasterIgnoreTargetedSlotDescriptor.kt
package com.ssblur.scriptor.word.descriptor.focus.inventory

import com.ssblur.scriptor.api.word.Descriptor
import com.ssblur.scriptor.helpers.targetable.InventoryTargetable
import com.ssblur.scriptor.helpers.targetable.Targetable
import com.ssblur.scriptor.word.descriptor.focus.FocusDescriptor

class CasterIgnoreTargetedSlotDescriptor: Descriptor(), FocusDescriptor {
  override fun cost() = Cost(0.0, COSTTYPE.ADDITIVE)

  override fun modifyFocus(targetable: Targetable): Targetable {
    if (targetable is InventoryTargetable) targetable.targetedSlot = -1
    return targetable
  }
}
